core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryActivity.kt

```diff
@@ -0,0 +1,141 @@
+import kotlinx.android.synthetic.main.layout_toolbar.toolbar
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.id
+import org.kiwix.kiwixmobile.core.R.string
+import org.kiwix.kiwixmobile.core.R2.id.menu_history_search
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.coreActivityComponent
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.viewModel
+import org.kiwix.kiwixmobile.core.history.HistoryAdapter.OnItemClickListener
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryAdapter2
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryDelegate.HistoryItemDelegate
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem
+import org.kiwix.kiwixmobile.core.history.adapter.HistoryListItem.HistoryItem
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.Filter
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.OnItemLongClick
+import org.kiwix.kiwixmobile.core.history.viewmodel.Action.OnSwitch
+import org.kiwix.kiwixmobile.core.history.viewmodel.HistoryViewModel
+import org.kiwix.kiwixmobile.core.history.viewmodel.State
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.NoResults
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.Results
+import org.kiwix.kiwixmobile.core.history.viewmodel.State.SelectionResults
+import org.kiwix.kiwixmobile.core.utils.SimpleTextListener
+import java.util.ArrayList
+import javax.inject.Inject
+
+class HistoryActivity : OnItemClickListener, BaseActivity() {
+
+  val activityComponent by lazy { coreActivityComponent }
+  private val historyList: List<HistoryListItem> = ArrayList()
+  private val fullHistory: List<HistoryListItem> = ArrayList()
+  val search = menu_history_search
+  private val deleteList: List<HistoryListItem> = ArrayList()
+  @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+  private lateinit var recyclerView: RecyclerView
+  private lateinit var historySwitch: Switch
+  private val historyViewModel by lazy { viewModel<HistoryViewModel>(viewModelFactory) }
+  private val compositeDisposable = CompositeDisposable()
+
+  private val historyAdapter: HistoryAdapter2 by lazy {
+    HistoryAdapter2(
+      HistoryItemDelegate(deleteList, this)
+    )
+  }
+
+  override fun injection(coreComponent: CoreComponent) {
+    coreComponent.activityComponentBuilder()
+      .activity(this)
+      .build()
+      .inject(this)
+    activityComponent.inject(this);
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_history)
+    setSupportActionBar(toolbar)
+    val actionBar = supportActionBar
+    actionBar?.setDisplayHomeAsUpEnabled(true)
+    actionBar?.setTitle(string.history)
+    recyclerView = recycler_view
```

Comment:
we don't need these variables for our views, using the properties from Android View Extension is enough, if you find the case annoying just rename them in xml

Reply:
Ah, much nicer!